packages/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -302,26 +303,25 @@ export function useGridDimensions(
   }, [apiRef, savedSize, updateDimensions]);
 
   const root = apiRef.current.rootElementRef.current;
-  const dimensions = apiRef.current.state.dimensions;
```

Comment:
I was under the impression this rule is not activated, as there are multiple instances where we do direct state access around the code base. Should it detect all the instances? 🤔

Reply:
I don't know, Matheus created it I think, since it's a rule that is mostly here for us, if it does not work maybe we can fix it or drop it :laughing: